### DISCUSSION TITLE
[General] zk init error

### DISCUSSION THREAD

**Discussion by tmors** [zkSync-Community-Hub/zksync-developers#779] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I want to build server-v2 and local-node images on my VM. I run these commands below and get an error msg 
```
git submodule update --init --recursive

cd ./zkstack_cli/zkstackup
./install --local
cd -
zkstackup --local

yarn install

make -C ./docker build-server-v2
ZKSYNC_HOME=`pwd` RUST_LOG=debug PATH=$ZKSYNC_HOME/bin:$PATH  zk 
ZKSYNC_HOME=`pwd` RUST_LOG=debug PATH=$ZKSYNC_HOME/bin:$PATH zk init
```
there is the error msg:

2024-11-19T02:11:57.867939Z ERROR zksync_node_framework::service: Task oneshot_runner failed: Oneshot task node_storage_initializer failed

Caused by:
    Base system contracts mismatched: From Config BaseSystemContractsHashes { bootloader: 0x010008c3be57ae5800e077b6c2056d9d75ad1a7b4f0ce583407961cc6fe0b678, default_aa: 0x0100055dba11508480be023137563caec69debc85f826cb3a4b68246a7cabe30, evm_emulator: None }, Calculated : BaseSystemContractsHashes { bootloader: 0x010008e15394cd83a8d463d61e00b4361afbc27c932b07a9d2100861b7d05e78, default_aa: 0x01000523eadd3061f8e701acda503defb7ac3734ae3371e4daf7494651d8b523, evm_emulator: None }

<img width="1493" alt="image" src="https://github.com/user-attachments/assets/eed355cf-9dc5-437d-903f-bfd78055bf0e">

**Comment by alhenaai** [comment]
It appears that the error you're encountering is due to a mismatch in the base system contracts during the initialization of your ZKsync node. This issue typically arises when the configuration files do not match the actual system contracts deployed or expected by the node framework.

Here are a few steps you can take to resolve this issue:

1. **Verify Configuration Files**: Ensure that the hashes in your configuration files match those expected by the system. You might need to update these hashes to reflect the correct versions of the contracts.

2. **Rebuild Images**: Sometimes, rebuilding the images can help ensure that all components are up to date. Run the build commands again to see if this resolves the issue.

3. **Check for Updates**: Make sure that you are using the latest versions of the ZKsync tools and dependencies. An update might have been released that fixes this issue.

4. **Consult Documentation**: For more detailed troubleshooting, refer to the ZKsync developers discussions on GitHub, where similar issues and their resolutions are discussed. You can find more information and ask questions there if needed. Here is the link to the [ZKsync developers discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

If these steps do not resolve the issue, you might want to provide more details about the configuration and the exact steps you are following so that further assistance can be tailored to your specific setup.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by MexicanAce** [comment, marked as answer]
Hi @tmors 👋 . I see you're using the `zk` CLI tool, I would recommend switching over to use the new `zkstack` CLI tool. So the equivalent command(s) to starting the server would be:

```sh
zkstack containers
zkstack ecosystem init --dev
zkstack server
```

[See the developer docs here](https://matter-labs.github.io/zksync-era/core/latest/guides/development.html) for more information about getting started with the `zkstack` CLI tool.

**Reply by tmors** [reply]
@MexicanAce 
but I need to build local-node, there is no build command about local-node on Makefile, so I have to run the command.
```
ZKSYNC_HOME=`pwd`  PATH=$ZKSYNC_HOME/bin:$PATH  zk  docker  build  local-node  --custom-tag  "test"
```
it would be error like this
<img width="1008" alt="image" src="https://github.com/user-attachments/assets/726c20a5-25fa-409c-89fe-f77f4a1a9d7a">

**Comment by thomas-nguy** [comment]
Maybe be worth to specify the branch/tag you are at.

@MexicanAce The zkstack command tool has been consistently failing on my machine as well and need to fix the hash manually (on v25.0.0 at least)
